apps/deploy-web/src/components/new-deployment/ManifestEdit.tsx

```diff
@@ -58,6 +60,34 @@ export const ManifestEdit: React.FunctionComponent<Props> = ({ editedManifest, s
   const sdlBuilderRef = useRef<SdlBuilderRefType>(null);
   const { minDeposit } = useChainParam();
 
+  const nextRouter = useNextRouter();
+  const { templateId } = nextRouter.query;
+
+  const fileUploadRef = useRef<HTMLInputElement>(null);
+
+  const handleFileChange = event => {
```

Comment:
I realise this is common in react world to name functions/callbacks this way. However for better readability functions should reflect what they are doing.  Something like this

```suggestion
  const propagateUploadedSdl = event => {
```

Feel free to come up with something better if you like so.

Just for example how this would look below on the input:

```typescript
<input type="file"  onChange={propagateUploadedSdl} />
```

This way it describes what happens and when. This reduces cognitive load of understanding code later on.